cmd/kube-apiserver/app/testing/testserver.go

```diff
@@ -174,6 +174,11 @@ func StartTestServer(t Logger, instanceOptions *TestServerInstanceOptions, custo
 
 	fs := pflag.NewFlagSet("test", pflag.PanicOnError)
 
+	// Intercept feature gate flags to steer tests toward SetFeatureGateDuringTest.
```

Comment:
is checking if `customFlags` contains `--feature-gates` or `--feature-gates=*` a simpler way of catching this